drivers/pds/targetcluster/targetcluster.go

```diff
@@ -48,6 +48,60 @@ type TargetCluster struct {
 	kubeconfig string
 }
 
+func (targetCluster *TargetCluster) IsLatestPDSHelm(helmChartversion string) (bool, error) {
+	cmd := "helm ls --all -n pds-system | tail -n+2 | awk '{print $9}' "
+	output, _, err := osutils.ExecShell(cmd)
+	if err != nil {
+		return false, err
+	}
+	log.InfoD("Installed PDS Helm chart version - %v", output)
+	_, pdsHelmVersion, _ := strings.Cut(output, "-")
+	_, helmVersion, _ := strings.Cut(pdsHelmVersion, "-")
+	helmVersion = strings.TrimSpace(helmVersion)
+	helmChartversion = strings.TrimSpace(helmChartversion)
+	log.Debugf("Installed PDS Helm version %s and helm chart version passed %s", helmVersion, helmChartversion)
+	return strings.EqualFold(helmVersion, helmChartversion), nil
+}
+
+func (targetCluster *TargetCluster) DeRegisterFromControlPlane() error {
```

Comment:
FYI: https://pds.docs.portworx.com/admin-guide/remove-cluster/remove-cluster-wipe-data/
Please follow the steps mentioned in above doc for clean de-registration.

Reply:
From the above docs we are missing only one step and that is deleing of CRDs and thats expected to stay post removal of pds helm charts, also we are immediately Installing PDS helm chart.
we can take this step while automating Deregistering target cluster testcase

Reply:
Please add a todo for same.

Reply:
Done